Creating generic mautrix bridge doc (#1912)
spantaleev/matrix-docker-ansible-deploy: docs/configuring-playbook-bridge-mautrix-Generic.md

# Setting up a Generic Mautrix Bridge (optional)

The playbook can install and configure bridges with mautrix (Currently twitter, facebook, instagram, signal, hangouts, googlechat)

You can see each bridge features at https://github.com/mautrix/SERVICENAME/blob/master/ROADMAP.md

To enable a bridge add

```yaml
matrix_mautrix_SERVICENAME_enabled: true
```

to your `vars.yml`

> There are some additional things you may wish to configure about the bridge before you continue.

Reviewer: ```suggestion
There are some additional things you may wish to configure about the bridge before you continue. Each bridge may have additional requirements besides `_enabled: true`. For example, the mautrix-telegram bridge (our documentation page about it is [here](configuring-playbook-bridge-mautrix-telegram.md)) requires the `matrix_mautrix_telegram_api_id` and `matrix_mautrix_telegram_api_hash` variables to be defined. Refer to each bridge's individual documentation page for details about enabling bridges.
```

> You can add

Reviewer: ```suggestion
You can add:
```

```yaml
matrix_admin: "@YOUR_USERNAME:{{ matrix_domain }}"
```

> to 'vars.yml' to configure a user as administrator for all bridges, or you prefer to do it bridge by bridge you can configure it with

Reviewer: ```suggestion
to `vars.yml` to **configure a user as an administrator for all bridges**.

**Alternatively** (more verbose, but allows multiple admins to be configured), you can do the same on a per-bridge basis with:
```

```yaml
matrix_mautrix_SERVICENAME_configuration_extension_yaml: |
  bridge:
    permissions:
      '@YOUR_USERNAME:{{ matrix_domain }}': admin
```

Encryption support is off by default. If you would like to enable encryption, add the following to your `vars.yml` file:
```yaml
matrix_mautrix_SERVICENAME_configuration_extension_yaml: |
  bridge:
    encryption:
      allow: true
      default: true
```

> Using both would look like

Reviewer: ```suggestion
You can only have one `matrix_mautrix_SERVICENAME_configuration_extension_yaml` definition in `vars.yml` per bridge, so if you need multiple pieces of configuration there, just merge them like this:
```

```yaml
matrix_mautrix_SERVICENAME_configuration_extension_yaml: |
  bridge:
    permissions:
      '@YOUR_USERNAME:{{ matrix_domain }}': admin
    encryption:
      allow: true
      default: true
```

> 

Reviewer: ```suggestion

## Setting the bot's username
```

It's helpful to add sections to this file, instead of having a wall of text with various things.

Feel free to do the same (adding sections) in other places in this file.

```yaml
matrix_mautrix_SERVICENAME_appservice_bot_username: "BOTNAME"
```

Can be used to set the username for the bridge.

> 

Reviewer: ```suggestion

## Discovering additional configuration options
```

^^ Adding another section.

please do so automatically, by enabling Shared Secret Auth

The bridge will automatically perform Double Puppeting if you enable [Shared Secret Auth](configuring-playbook-shared-secret-auth.md) for this playbook by adding

```yaml
matrix_synapse_ext_password_provider_shared_secret_auth_enabled: true
matrix_synapse_ext_password_provider_shared_secret_auth_shared_secret: YOUR_SHARED_SECRET_GOES_HERE
```

You should generate a strong shared secret with a command like this: pwgen -s 64 1

This is the recommended way of setting up Double Puppeting, as it's easier to accomplish, works for all your users automatically, and has less of a chance of breaking in the future.

## Usage

You then need to start a chat with `@SERVICENAMEbot:YOUR_DOMAIN` (where `YOUR_DOMAIN` is your base domain, not the `matrix.` domain). 

Send `login ` to the bridge bot to get started You can learn more here about authentication from the bridge's official documentation on Authentication https://docs.mau.fi/bridges/python/SERVICENAME/authentication.html  .

If you run into trouble, check the [Troubleshooting](#troubleshooting) section below.

## Troubleshooting

> Please see SERVICENAME's individual doc page for troubleshooting information.

Reviewer: Saying `SERVICENAME` is not very useful here.

```suggestion
For troubleshooting information with a specific bridge, please see the playbook documentation about it (some other document in in `docs/`) and the upstream ([mautrix](https://github.com/mautrix)) bridge documentation for that specific bridge.

Reporting bridge bugs should happen upstream, in the corresponding mautrix repository, not to us.
```
